### ISSUE TITLE
GridFieldAddNewInlineButton throws exception in SS4 (even when editable columns is installed)

### ISSUE THREAD

**Issue by mlewis-everley** [symbiote/silverstripe-gridfieldextensions#230]
Example Code:

`````
<?php

namespace SilverCommerce\OrdersAdmin\Model;

use SilverStripe\ORM\DataObject;
use SilverStripe\Forms\FieldList;
use SilverStripe\Forms\GridField\GridFieldAddNewButton;
use Symbiote\GridFieldExtensions\GridFieldEditableColumns;
use Symbiote\GridFieldExtensions\GridFieldAddNewInlineButton;

class LineItem extends DataObject
{
    private static $has_many = [
        "Customisations" => LineItemCustomisation::class
    ];

    public function getCMSFields()
    {
        $fields = parent::getCMSFields();

        // Change unlink button to remove on customisation
        $custom_field = $fields->dataFieldByName("Customisations");

        if ($custom_field) {
            $config = $custom_field->getConfig();
            $config
                ->removeComponentsByType(GridFieldAddNewButton::class)
                ->removeComponentsByType(GridFieldDataColumns::class)
                ->removeComponentsByType(GridFieldEditButton::class)
                ->removeComponentsByType(GridFieldAddExistingAutocompleter::class)
                ->addComponents(
                    new GridFieldEditableColumns(),
                    new GridFieldAddNewInlineButton()
                );
        }

        return $fields;
    }
}
`````

This throws an exception:

`[Emergency] Uncaught Exception: Inline adding requires the editable columns component`

The same code works as expected in SS3.

Dropping a Debug::show() at: https://github.com/symbiote/silverstripe-gridfieldextensions/blob/master/src/GridFieldAddNewInlineButton.php#L96

Shows that editable returns false, but dropping the same Debug::show() at the end of my if statement shows that GridFieldEditableColumns has been added to the config.

Any ideas on why this is happening? I am totally stumped :-o

**Comment by dhensby** [comment]
is this still a problem?

**Comment by mlewis-everley** [comment]
It appears to be, have you pushed an update or something?

Also, should mention:

SS Version: 4.0.0
PHP: 7.1.11

**Comment by robbieaverill** [comment]
I feel like this was fixed but not tagged yet. Have you tried using 3.0.x-dev?

**Comment by mlewis-everley** [comment]
OK, this appears to be fixed in 3.0.x-dev. Any chance of a new tag?

**Comment by dhensby** [comment]
@mlewis-everley sorry for the confusion, I hadn't realised how new the ticket was when i commented (as I was going through old tickets at the time). glad it's fixed. I'll leave tagging up to @robbieaverill
